src/components/MusicForm/MusicForm.spec.js

```diff
@@ -0,0 +1,42 @@
+import { render, screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import MusicForm from './MusicForm'
+import { MemoryRouter } from 'react-router-dom'
+
+describe('MusicForm', () => {
+  it('renders a form with two inputs and a button as link', () => {
+    render(
+      <MemoryRouter initialEntries={['/']}>
+        <MusicForm />
+      </MemoryRouter>
+    )
+
+    expect(screen.getByLabelText('Energy')).toBeInTheDocument()
+    expect(screen.getByLabelText('Genre')).toBeInTheDocument()
+    expect(screen.getByRole('link')).toBeInTheDocument()
+  })
+
+  it('renders two required input fields', () => {
+    render(
+      <MemoryRouter initialEntries={['/music']}>
+        <MusicForm />
+      </MemoryRouter>
+    )
+
+    expect(screen.getByLabelText('Energy')).toBeRequired()
+    expect(screen.getByLabelText('Genre')).toBeRequired()
+  })
+
+  it('selects the right option', () => {
+    const callback = jest.fn()
+    render(
+      <MemoryRouter initialEntries={['/music']}>
+        (<MusicForm onSubmit={callback} onCreatePlaylist={callback} />)
```

Comment:
You don't have an `onSubmit` property in your `MusicForm`. This doesn't do anything. You only want to test if the `onCreatePlaylist` function is called when you submit the form. 
```suggestion
        (<MusicForm onCreatePlaylist={callback} />)
```

Reply:
Yes, you are right. I added your suggestions. Thanks!